core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
Actually, it _must_ be shutdown after termination, as RPCs are still functioning after `shutdown()` is called (just no new ones are permitted). So we will need deadline support even when shutdown. When `terminated` it is guaranteed there are no RPCs in progress, so it is safe to release this scheduler.